spotfinder/spotfinder.cc

```diff
@@ -330,6 +308,25 @@ int main(int argc, char **argv) {
     float dmin = parser.get<float>("dmin");
     float dmax = parser.get<float>("dmax");
 
+    DispersionAlgorithm dispersion_algorithm;
+    {  // Parse the algorithm input
+        std::string dispersion_algorithm_str = parser.get<std::string>("algorithm");
+        std::transform(dispersion_algorithm_str.begin(),
+                       dispersion_algorithm_str.end(),
+                       dispersion_algorithm_str.begin(),
+                       ::tolower);
+        if (dispersion_algorithm_str == "dispersion") {
+            dispersion_algorithm = DispersionAlgorithm::DISPERSION;
+        } else if (dispersion_algorithm_str == "dispersion_extended") {
+            dispersion_algorithm = DispersionAlgorithm::DISPERSION_EXTENDED;
+        } else {
+            print("Error: Unknown dispersion algorithm '{}'\n",
+                  dispersion_algorithm_str);
+            std::exit(1);
+        }
+        print("Algorithm: {}\n", styled(dispersion_algorithm_str, fmt_green));
+    }
+
```

Comment:
Move this to a separate `parse_algorithm(std::String named)`

Reply:
I put it in [a struct for parsing and access](https://github.com/DiamondLightSource/fast-feedback-service/pull/1/commits/acdb147642ef4847d1ab5f2db7d5fd80c06954d3), hopefully encapsulating the whole process a little better